projects/plugins/boost/app/modules/cache/storage/Storage.php

```diff
@@ -9,4 +9,7 @@ interface Storage {
 
 	public function write( $request_uri, $parameters, $data );
 	public function read( $request_uri, $parameters );
+	public function invalidate( $request_uri );
+	public function invalidate_single_visitor( $request_uri, $parameters );
+	public function invalidate_home_page( $dir );
```

Comment:
Because this is an interface, I would caution against requiring the Storage to know about specifics of what it's storing- "single visitor" or "home_page", that should probably be done at the implementation level.

Looking at the surrounding code, It looks to me like the `invalidate` interface should be `invalidate( $request_uri, $parameters)` and the rest should be handled by the implementation instead.

But I haven't followed along the developments here and I'm on SR right now, so just adding this as a comment here.

Reply:
I understand where you're coming from, and I agree that we should think about what each class is responsible for.

But in this specific instance, I don't think you're correct. The Storage class is specific to caching, and exists to understand how to store data indexed by its metadata, in a format where it can have segments of its cache wiped based on their relationships.

I actually do think it's important for the Storage interface here to understand the semantics of the data it's storing, because it is so very, very specific to how it is to be stored.